### PR TITLE
Grant medical HUD to Artificers

### PR DESCRIPTION
**Pull request by Cenrus**
## About The Pull Request

Adds a medical HUD to Artificer constructs, allowing them to see the health of their fellow constructs at a glance.

## Why It's Good For The Game

Helps Artificers know which of their fellow constructs are damaged, so that they can do their support duties more efficiently. It's hard to know who is calling for repairs in a crowded cult fortress.

## Changelog
:cl: The Space Wizard Federation
add: Artificers now have medical HUDs.
/:cl:

### CODE REVIEW

**Comment by coiax** [comment]
At the risk of making the PR way more complex, could you give the medical HUD to only cultists and constructs? I am not sure if the current HUD system supports that level of complexity though.

**Comment by AlinaStarkova** [comment]
> 
> 
> At the risk of making the PR way more complex, could you give the medical HUD to only cultists and constructs? I am not sure if the current HUD system supports that level of complexity though.

this gives it only to the Artificers, since Artificers are intended to heal other constructs, regardless if they are cult artificers, chaplain artifiers, or wizard artificers. cultists can get their own med huds using their blindfolds

**Comment by Jared-Fogle** [comment]
@AlinaStarkova he means making it so they only see the health huds of other cultists. it's possible but probably out of scope for this pr

**Comment by AlinaStarkova** [comment]
> @AlinaStarkova he means making it so they only see the health huds of other cultists. it's possible but probably out of scope for this pr

ahhh, the wording threw me off, as he said "give the health hud" rather than show the health.

File: code/modules/mob/living/simple_animal/constructs.dm [hunk 1]
```diff
@@ -276,6 +276,12 @@
 						and shells to place those soulstones into.</b>"
 	can_repair_constructs = TRUE
 	can_repair_self = TRUE
+	var/health_hud = DATA_HUD_MEDICAL_ADVANCED
```
Comment (Jared-Fogle) [inline code/modules/mob/living/simple_animal/constructs.dm:279]: Document this--https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md

**Comment by ATH1909** [comment]
we should probably consider giving a few other mobs appropriate HUDs, like hivebot mechanics (diagnostic HUDs) and (s)laughter demons (health HUDs so that they can tell who is vulnerable to being picked off).

**Review by Mothblocks — APPROVED** [review]

**Pull request merged**